sonar-slang-plugin/src/main/java/com/sonarsource/slang/kotlin/KotlinTreeVisitor.java

```diff
@@ -330,8 +331,18 @@ private static boolean isError(PsiElement element) {
 
   private static boolean isLiteral(PsiElement element) {
     return element instanceof KtConstantExpression
-      || element instanceof KtLiteralStringTemplateEntry
+      || isSimpleStringLiteral(element);
+  }
+
+  private static boolean isSimpleStringLiteral(PsiElement element) {
+    return element instanceof KtStringTemplateExpression
+      && Stream.of(((KtStringTemplateExpression) element).getEntries()).allMatch(KotlinTreeVisitor::isSimpleStringLiteralEntry);
+  }
+
+  private static boolean isSimpleStringLiteralEntry(PsiElement element) {
+    return element instanceof KtLiteralStringTemplateEntry
       || element instanceof KtEscapeStringTemplateEntry
       || (element instanceof KtStringTemplateExpression && !((KtStringTemplateExpression) element).hasInterpolation());
```

Comment:
I must say that I don't understand everything, but removing this line does not impact unit tests...

Reply:
Right. This was a mistake on my part during refactoring.
Small explanation: the part `(element instanceof KtStringTemplateExpression && !((KtStringTemplateExpression) element).hasInterpolation());` does the same thing as `Stream.of(((KtStringTemplateExpression) element).getEntries()).allMatch(KotlinTreeVisitor::isSimpleStringLiteralEntry);` and it is not needed in that method. 
As here we only check the entries of the String template that are string literals (i.e. we still want to differentiate in the native `StringTemplate` element its children).
Updated it before merge